src/Rules/Registry.php

```diff
@@ -33,7 +33,7 @@ public function getRules(string $nodeType): array
 			$rules = [];
 			foreach ($nodeTypes as $nodeType) {
 				foreach ($this->rules[$nodeType] ?? [] as $rule) {
-					$rules[get_class($rule)] = $rule;
+					$rules[spl_object_hash($rule)] = $rule;
```

Comment:
Hi, I don't think this is needed at all. Can you change this line to `$rules[] = $rule;` and test it? Thanks.

Reply:
Yep, hash map is not necessary here, removed.